cartographer/common/internal/testing/thread_pool_for_testing.h

```diff
@@ -34,7 +34,14 @@ class ThreadPoolForTesting : public ThreadPoolInterface {
   ThreadPoolForTesting();
   ~ThreadPoolForTesting();
 
-  void Schedule(const std::function<void()> &work_item) override;
+  void NotifyReady(Task* task) EXCLUDES(mutex_) override {
```

Comment:
Rename to NotifyDependenciesCompleted and make private since this is not part of the public interface. Make Task a friend of ThreadPool.

Reply:
Done. Actually, friend PoseGraphInterface does not have effect on the children, so I needed to add access functions in the base class.